tools/rapids-retry

```diff
@@ -26,25 +29,63 @@
 #                          seconds, to wait between retries.
 #                          Default is a 10 second sleep
 #
+# Additional options:
+#
+# --quiet                - when specified as the first argument after the command,
+#                          captures the command's output instead of sending it to stdout.
+#                          Logger messages are redirected to stderr.
+#                          Returns the output of the command on success.
+#
 function rapids-retry {
+    # Handle the --quiet flag if it's the first argument
+    quiet=false
+    if [[ "$1" == "--quiet" ]]; then
+        quiet=true
+        shift
+    fi
     command=$1
     shift
     max_retries=${RAPIDS_RETRY_MAX:=3}
     retries=0
     sleep_interval=${RAPIDS_RETRY_SLEEP:=10}
 
-    ${command} "$@"
-    retcode=$?
-    while (( retcode != 0 )) && \
-          (( retries < max_retries )); do
-      ((retries++))
-      rapids-logger "rapids-retry: retry ${retries} of ${max_retries} | exit code: (${retcode}) -> sleeping for ${sleep_interval} seconds..."
-      sleep "${sleep_interval}"
-      rapids-logger "rapids-retry: sleep done -> retrying..."
-
-      ${command} "$@"
-      retcode=$?
+    local output
+    local retcode
+
+    if [[ "$quiet" == "true" ]]; then
+        output=$("${command}" "$@")
+        retcode=$?
+    else
+        ${command} "$@"
+        retcode=$?
+    fi
+
+    # Choose rapids-logger FD based on quiet mode: stderr (FD 2) if quiet, stdout (FD 1) otherwise
+    logger_fd="1"
+    if [[ "$quiet" == "true" ]]; then
+        logger_fd="2"
+    fi
```

Comment:
I think this needs to come before the block above in lines 55-61, since the command runs there first before going to the retry loop to run again.

Reply:
If it passes on the first attempt, there are no calls to `rapids-logger`. The `logger_fd` is only needed in the loop. However, I agree that it makes sense to structure it so that the first attempt and the looped attempts are next to each other. Updated in 857fb9480649c36290f88002734723e2ba6bbcbb.